modules/dfpAdServerVideo.js

```diff
@@ -249,6 +254,68 @@ function getCustParams(bid, options, urlCustParams) {
   return encodedParams;
 }
 
+async function getVastForLocallyCachedBids(gamVastWrapper, localCacheMap) {
+  try {
+    const xmlUtil = XMLUtil();
+    const xmlDoc = xmlUtil.parse(gamVastWrapper);
+    const vastAdTagUriElement = xmlDoc.querySelectorAll(VAST_TAG_URI_TAGNAME)[0];
+
+    if (!vastAdTagUriElement || !vastAdTagUriElement.textContent) {
+      return gamVastWrapper;
+    }
+
+    const uuidExp = new RegExp(`.*(?:([A-Fa-f0-9]{8}-(?:[A-Fa-f0-9]{4}-){3}[A-Fa-f0-9]{12}).*)+`, 'gi');
+    const matchResult = Array.from(vastAdTagUriElement.textContent.matchAll(uuidExp))[0];
+    const uuid = Array.from(localCacheMap.keys()).find((videoCacheKey) => matchResult.includes(videoCacheKey));
+
+    if (uuid) {
+      const blobUrl = localCacheMap.get(uuid);
+      const base64BlobContent = await getBase64BlobContent(blobUrl);
+      const cdata = xmlDoc.createCDATASection(base64BlobContent);
+      vastAdTagUriElement.textContent = '';
+      vastAdTagUriElement.appendChild(cdata);
+      return xmlUtil.serialize(xmlDoc);
+    } else {
+      throw new Error('Unable to find uuid in ad tag');
+    }
```

Comment:
This can just check if the uuid is in the map no? no need to iterate over it. And when it's not in the map, or when it finds multiple that are in in it, it should return the unmodified VAST - not throw an error.